v-next/hardhat-keystore/src/internal/hook-handlers/configuration-variables.ts

```diff
@@ -0,0 +1,50 @@
+import type { KeystoreLoader } from "../types.js";
+import type { ConfigurationVariable } from "@ignored/hardhat-vnext/types/config";
+import type {
+  ConfigurationVariableHooks,
+  HookContext,
+} from "@ignored/hardhat-vnext/types/hooks";
+
+import { FileManagerImpl } from "../loaders/file-manager.js";
+import { KeystoreFileLoader } from "../loaders/keystore-file-loader.js";
+
+export default async (): Promise<Partial<ConfigurationVariableHooks>> => {
+  // Use a cache with hooks since they may be called multiple times consecutively.
+  let keystoreLoader: KeystoreLoader | undefined;
```

Comment:
We can the loader here as shared state between runs of the handler.

It can't be initialized here as we need access to the hook context (for the user interruptions).

Reply:
Note there is no caching in the tasks. I assume we don't expect running of the tasks from scripts?

Reply:
This works as expected. The reason we export factories, and not the hook handlers directly, is precisely to be able to do this.